nuxeo-features/nuxeo-platform-userworkspace/nuxeo-platform-userworkspace-core/src/main/java/org/nuxeo/ecm/platform/userworkspace/core/listener/InvalidateUserWorkspacesListener.java

```diff
@@ -41,7 +42,10 @@ public void handleEvent(Event event) {
         if (!(ec instanceof DocumentEventContext)) {
             return;
         }
-        if (!DocumentEventTypes.DOCUMENT_REMOVED.equals(event.getName())) {
+        String evtName = event.getName();
+        if (!DocumentEventTypes.DOCUMENT_REMOVED.equals(evtName)
+                && !TrashService.DOCUMENT_TRASHED.equals(evtName)
+                && !TrashService.DOCUMENT_UNTRASHED.equals(evtName)) {
```

Comment:
This kind of check is redundant given that there's already a list of events to filter on in the XML.